AI/VCAI/Fuzzy.cpp

```diff
@@ -416,37 +425,47 @@ void FuzzyHelper::initVisitTile()
 float FuzzyHelper::evaluate (Goals::VisitTile & g)
 {
 	//we assume that hero is already set and we want to choose most suitable one for the mission
-	if (!g.hero)
+	if(!g.hero)
 		return 0;
 
 	//assert(cb->isInTheMap(g.tile));
 	float turns = 0;
 	float distance = CPathfinderHelper::getMovementCost(g.hero.h, g.tile);
-	if (!distance) //we stand on that tile
+	if(!distance) //we stand on that tile
 		turns = 0;
 	else
 	{
-		if (distance < g.hero->movement) //we can move there within one turn
+		if(distance < g.hero->movement) //we can move there within one turn
 			turns = (fl::scalar)distance / g.hero->movement;
 		else
 			turns = 1 + (fl::scalar)(distance - g.hero->movement) / g.hero->maxMovePoints(true); //bool on land?
 	}
 
 	float missionImportance = 0;
-	if (vstd::contains(ai->lockedHeroes, g.hero))
+	if(vstd::contains(ai->lockedHeroes, g.hero))
 		missionImportance = ai->lockedHeroes[g.hero]->priority;
 
 	float strengthRatio = 10.0f; //we are much stronger than enemy
 	ui64 danger = evaluateDanger (g.tile, g.hero.h);
-	if (danger)
+	if(danger)
 		strengthRatio = (fl::scalar)g.hero.h->getTotalStrength() / danger;
 
+	float tilePriority = 0;
+	if(g.objid == -1)
+		vt.estimatedReward->setEnabled(false);
+	else if(g.objid == Obj::TOWN)
+	{
+		vt.estimatedReward->setEnabled(true);
+		tilePriority = 5;
+	}
+		

```

Comment:
And here. Eventually GetObj should take all the functionality of visitTile and add this additional rule for object evaluation. This implementation is very messy.